### PR TITLE
feat(platform): Create ui link for resend otp

### LINKED ISSUE #479: PLATFORM: Implement resend OTP flow
## Description
In the current `auth` flow, we don't have a resent OTP button, meaning if the OTP times out, you don't have a way to get the OTP. You need to implement the resent OTP in the `/auth/otp`.

## Scope of Work
- You need to implement the UI for resend OTP flow in the [`apps/platform/src/app/auth/otp/page.tsx`](https://github.com/keyshade-xyz/keyshade/blob/develop/apps/platform/src/app/auth/otp/page.tsx) page.
- When the resent OTP button is clicked, you need to call `{BASE_URL}/api/auth/send-otp/:email` endpoint
## URLs you need
- Figma UI for resend OTP flow [link](https://www.figma.com/design/Du3qt5NWbbnfMgtU5CQYCU/Keyshade-UI?node-id=862-489&node-type=canvas&t=PmGAEhY7HdEKfFGT-0)
- Postman URL for `{BASE_URL}/api/auth/send-otp/:email` [link](https://www.postman.com/keyshade/workspace/keyshade/api/a31bdb66-69e3-469b-afb4-f2051385e634/request/32733901-3db2069c-6157-4ee2-a64f-f053f380646a?action=share&source=copy-link&creator=32744470&active-environment=8fc04ef8-aeca-488d-828a-095b724fb9de)

### PR DESCRIPTION
**Pull request by Prakhargarg-2010196**
## Description

Adding the link to platform ui for resending otp <br />

Fixes #479 

## Screenshots of relevant screens
resend-otp-2024-10-14_15 26 04

## Developer's checklist

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-check on my work

**If changes are made in the code:**

- [x] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [x] My changes in code generate no new warnings
- [x] I have added relevant screenshots in my PR
- [x] There are no UI/UX issues(I suppose)

___

### **PR Type**
Enhancement

___

### **Description**
- Implemented the resend OTP functionality in the authentication page to improve user experience.
- Added a new function `handleResendOtp` to manage the resend OTP process, including API call and error handling.
- Introduced a loading state and UI feedback for the resend OTP action, including success and error notifications using toast.
- Enhanced the user interface with a "Resend OTP" button and loading indicator to guide users through the process.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>page.tsx</strong><dd><code>Implement resend OTP functionality in authentication page</code></dd></summary>
<hr>

apps/platform/src/app/auth/otp/page.tsx

<li>Added state management for resend OTP loading status.<br> <li> Implemented <code>handleResendOtp</code> function to handle OTP resend logic.<br> <li> Added UI elements for the resend OTP button with loading indicator.<br> <li> Integrated toast notifications for success and error scenarios.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/489/files#diff-3ae286985cfeac090cabd429156a45eeaae5875dc9d9e23d93549e3b289387ba">+45/-1</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: apps/platform/src/app/auth/otp/page.tsx [hunk 1]
```diff
@@ -88,7 +89,28 @@ export default function AuthOTPPage(): React.JSX.Element {
...
+      } else if (response.ok)
+        toast.success('OTP successfully sent to your email')
+      setIsLoadingRefresh(false)
+    } catch (error) {
+      // eslint-disable-next-line no-console -- we need to log the error
+      console.error(`Failed to send OTP: ${error}`)
+      toast.error("Couldn't send OTP .")
+      setIsLoadingRefresh(false)
+    }
+  }
```
Comment (kriptonian1) [inline apps/platform/src/app/auth/otp/page.tsx:113]: Can you organize the API logic in `apps/platform/src/lib/api-functions` rest looks good

**Review by kriptonian1 — CHANGES_REQUESTED** [review]
Everything looks good, just you need to organise the API call in the right file

**Comment by Prakhargarg-2010196** [comment]
> Everything looks good, just you need to organise the API call in the right file

Can you please point that file if possible it would be helpful thanks.

File: apps/platform/src/app/auth/otp/page.tsx (hunk 1 above)
Comment (rajdip-b) [reply apps/platform/src/app/auth/otp/page.tsx:113]: Don't use your own calls to the API. Rather use this: https://github.com/keyshade-xyz/keyshade/blob/5884833fd7c36e0021ddc7e50d06fe3374e90c15/packages/api-client/src/controllers/user.ts#L55-L65

You can check the usage of these in any of the CLI files: https://github.com/keyshade-xyz/keyshade/tree/develop/apps/cli/src/commands

**Comment by kriptonian1** [comment]
> > Everything looks good, just you need to organise the API call in the right file
> 
> Can you please point that file if possible it would be helpful thanks.

I have mentioned the file in the review

File: apps/platform/src/app/auth/otp/page.tsx (hunk 1 above)
Comment (Prakhargarg-2010196) [reply apps/platform/src/app/auth/otp/page.tsx:113]: > resendEmailChangeOTP

I got the point of the controller code from api-client but not the second thing you said
> You can check the usage of these in any of the CLI files: https://github.com/keyshade-xyz/keyshade/tree/develop/apps/cli/src/commands

Comment (rajdip-b) [reply apps/platform/src/app/auth/otp/page.tsx:113]: Oh, I meant that you can check any of the files in the `commands` directory to understand how to use these controllers.

**Comment by rajdip-b** [comment]
> > > Everything looks good, just you need to organise the API call in the right file
> > 
> > 
> > Can you please point that file if possible it would be helpful thanks.
> 
> I have mentioned the file in the review

We shouldn't be using a separate api function folder, we already have a dedicated package for that.

**Comment by kriptonian1** [comment]
> > > > Everything looks good, just you need to organise the API call in the right file
> > > 
> > > 
> > > Can you please point that file if possible it would be helpful thanks.
> > 
> > 
> > I have mentioned the file in the review
> 
> We shouldn't be using a separate api function folder, we already have a dedicated package for that.

Looks like I need to migrate some stuffs

File: apps/platform/src/app/auth/otp/page.tsx (hunk 1 above)
Comment (Prakhargarg-2010196) [reply apps/platform/src/app/auth/otp/page.tsx:113]: So i have to write a new controller for auth and utilise as a controller instance if I am getting it right? 

Is there a controller instance file similar instance to https://github.com/keyshade-xyz/keyshade/blob/develop/apps/cli/src/util/controller-instance.ts or I need to create one for the frontend of platform?

**Comment by rajdip-b** [comment]
@kriptonian1 @Prakhargarg-2010196 lmk when this is ready

**Comment by Prakhargarg-2010196** [comment]
> @kriptonian1 @Prakhargarg-2010196 lmk when this is ready

I am working on the changes but was bit confused 
Two api-client.ts files present one in the `apps/platform/src/lib/api-client.ts` and another where you suggested in packages
So i need to use which one @kriptonian1? And if so shall the api-client in platform be deleted ?

**Comment by rajdip-b** [comment]
Let the api-client.ts file be. You would need to use the one from `api-client` package.

File: apps/platform/src/app/auth/otp/page.tsx [hunk 2]
```diff
@@ -8,6 +8,8 @@ import Cookies from 'js-cookie'
 import { toast } from 'sonner'
 import { LoadingSVG } from '@public/svg/shared'
 import { KeyshadeBigSVG } from '@public/svg/auth'
+import type { ClientResponse } from '@keyshade/api-client/dist/src/types/index.types'
+import type { ResendOTPRequest } from '@keyshade/api-client/dist/src/types/auth.types'
```
Comment (rajdip-b) [inline apps/platform/src/app/auth/otp/page.tsx:12]: I don't think we should allow this import style. The types should come from `@keyshade/api-client/types/<type file>`

Comment (Prakhargarg-2010196) [reply apps/platform/src/app/auth/otp/page.tsx:12]: yup i wanted that manner only but this was just a workaround to show you how that  2nd solution would work which I mentioned from stackoverflow.

**Comment by Prakhargarg-2010196** [comment]
> @Prakhargarg-2010196 can you please configure ESLint on your device to use the config from keyshade? There are linting errors

That's what I was saying the errors are not being thrown locally when i run `pnpm run lint` doesn't shows any errors on local. 
image

Can you tell me some way to know which config is being consumed

File: apps/platform/src/app/auth/otp/page.tsx (hunk 2 above)
Comment (Prakhargarg-2010196) [reply apps/platform/src/app/auth/otp/page.tsx:12]: the issue is @keyshade/api-client is a package and points to `dist/src/index.js` in the api-client when we import it as written in the `package.json`. And hence the syntax suggested by you would work only if the some different path is assigned there or by some strategy. Any Suggestions?

**Comment by rajdip-b** [comment]
Okay got it. I'll switch to your PR and check it manually

**Comment by rajdip-b** [comment]
@kriptonian1 if you can lend me a hand then it would be awesome

**Comment by rajdip-b** [comment]
@Prakhargarg-2010196 I have relaxed the restrictions on ESLint

**Comment by rajdip-b** [comment]
You can ignore the lint failure, that's probably false.

**Comment by Prakhargarg-2010196** [comment]
> You can ignore the lint failure, that's probably false.

Then merge can be done right ?

**Review by rajdip-b — APPROVED** [review]
LGTM

**Comment by rajdip-b** [comment]
> > You can ignore the lint failure, that's probably false.
> 
> Then merge can be done right ?
Once @kriptonian1 drops his review, we can

**Review by kriptonian1 — APPROVED** [review]

**Pull request merged**